### ISSUE TITLE
Malformed ContentFeatures.DLNA.ORG header sent to renderer

### ISSUE THREAD

**Issue by master-nevi** [UniversalMediaServer/UniversalMediaServer#493]
This is my 3rd attempt trying to report and provide a fix for this bug, I hope somebody sees this.

Basically when there's a 3rd party control point coordinating the render and server, the DLNA-compliant renderer fails to request a video due to a malformed header coming back from UMS which contains a mixture of "null" strings and is partially incorrect in format.

I've already debugged the server and put together a code fix here: e201a4eb46801138aafec01f9a53a82d2be80f68

Basically there's a method called getDidlString() which, although it's a getter, oddly sets the instance variables "dlnaspec" and "dlnaOrgOpFlags" which are needed to ensure getDlnaContentFeatures() creates and returns the correct header string. So I had to call it before the return. Having a getter set instance variables is certainly a code smell to me but I didn't want to attempt a refactor without somebody agreeing with the bug and willing to accept the pull request.

**Comment by skeptical** [comment]
Thanks, sorry for the unintended oversight in not responding sooner :).  Seems like a valid bug and reasonable patch, and agree it would probably be better resolved outside the getter.

The fact is the codebase is inherited by us, and there are a lot modularity issues like this, partly because of the original coding style where many separate tasks tend to be interleaved in mile-long functions, and I guess also because later fixes have tended to avoid refactoring so as not to risk breaking anything.

**Comment by ExSport** [comment]
Seems this one should fix one problem mentioned by me here (missing FLAGS only): https://github.com/UniversalMediaServer/UniversalMediaServer/issues/494

**Comment by master-nevi** [comment]
Cool, I appreciate you looking at this. Here is my previous attempt at refactoring the problem area: https://github.com/UniversalMediaServer/UniversalMediaServer/commit/82f48e0efaa778707daed94ca7396ab9fcc68e5f

The only issue is that it's a refactor from last Nov so it may be missing other fixes. I could apply it to a more recent version such as the above fix if you agree with the changes. I basically get rid of the "dlnaspec" and "dlnaOrgOpFlags" string ivars and made the methods getDlnaOrgPnFlags() and getDlnaOrgOpFlags() for creating their values (dlnaspec was renamed to dlnaOrgPnFlags); this was especially necessary since both require the RendererConfiguration in order for both to be created anyways. As a result and added bonus, this change cuts the mile-long method getDidlString() in half as a large portion of that method was used to generate those two values.

**Comment by skeptical** [comment]
Great, if you make 82f48e0 mergeable I'll put it in.

**Comment by master-nevi** [comment]
Done and done: see my pull request here: https://github.com/UniversalMediaServer/UniversalMediaServer/pull/496
